static/js/compose_ui.js

```diff
@@ -130,3 +143,29 @@ export function wrap_text_with_markdown(textarea, prefix, suffix) {
         textarea.range(range.start, range.end).range(prefix + range.text + suffix);
     }
 }
+
+export function make_compose_box_full_size() {
+    set_full_size(true);
+
+    // The autosize should be destroyed for the full size compose
+    // box else it will interfare and shrink its size accordingly.
+    autosize.destroy($("#compose-textarea"));
+
+    $("#compose").addClass("compose-fullscreen");
+
+    $(".collapse_composebox_button").show();
+    $(".expand_composebox_button").hide();
+}
+
+export function make_compose_box_original_size() {
+    set_full_size(false);
+
+    $("#compose").removeClass("compose-fullscreen");
+
+    // Again initialise the compose textarea as it was destroyed
+    // when compose box was made full screen
+    autosize($("#compose-textarea"));
+
+    $(".collapse_composebox_button").hide();
+    $(".expand_composebox_button").show();
```

Comment:
I added a line in both of these functions to move focus back to the compose box, since that's almost always what you want.

Reply:
Great!